src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -450,6 +462,107 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
     }
   }
 
+  /**
+    * Collect data from Blocks Table, Operations Table, and the Tezos RPC,
+    * return the appropriate list of VoteAggregates, and write them to the Votes table.
+    * @param ex the needed ExecutionContext to combine multiple database operations
+    * @return a future result of the number of rows stored to db, if supported by the driver
+    */
+  private[this] def processTezosVotes(): Future[Done] = {
+    import slick.jdbc.PostgresProfile.api._
+    implicit val mat = ActorMaterializer()
+
+    logger.info("Processing latest Tezos votes data...")
+
+    /**
+      * * Helper function, takes a sequence of GovernanceData, separates them by ballot types.
+      * @param votingData
+      * @return
+      */
+    def partitionByVote(votingData: Seq[VotingData]): (Seq[VotingData], Seq[VotingData], Seq[VotingData]) = {
+      val (yays, notYays) = votingData.partition(x => x.ballot.contains("yay"))
```

Comment:
I guess here you can `groupBy` and collect the values of the returned map in a single pass

Reply:
used groupBy and foldLeft, not sure what did you mean by using collect?

Reply:
Oh, I meant simply  `groupBy(...).values` 😁

Reply:
I'm not sure if I understand your approach correctly, but I fee like it would fail on edge cases

Reply:
I think I didn't properly evaluate what the algorithm is doing, let's ignore my comments.